hcl2template/types.build.go

```diff
@@ -135,11 +161,11 @@ func (p *Parser) decodeBuildConfig(block *hcl.Block, cfg *PackerConfig) (*BuildB
 			diags = append(diags, &hcl.Diagnostic{
 				Severity: hcl.DiagError,
 				Summary:  "Invalid " + sourceLabel + " reference",
-				Detail: "A " + sourceLabel + " type is made of three parts that are" +
+				Detail: "A " + sourceLabel + " type is made of two or three parts that are" +
```

Comment:
So in actuality it is not necessarily three parts indeed! You can leave out the `source` part from the reference in the `sources` attribute, so that effectively becomes `type.name`.